### ISSUE TITLE
[Validator] Incorrect EmailValidator deprecation when using `Validation::createIsValidCallable()`

### ISSUE THREAD

**Issue by mhujer** [symfony/symfony#50217]
### Symfony version(s) affected

6.2.10

### Description

When using `Validation::createIsValidCallable()`, the validator is created in `ConstraintValidatorFactory` with `new $class()` so it incorrectly emits the deprecation `The "loose" mode is deprecated` even though the constraint is correctly configured with HTML5 mode.

There is a workaround available for `Validation::createValidatorBuilder()` (added in https://github.com/symfony/symfony/pull/48644), but not for `Validation::createIsValidCallable()`.

### How to reproduce

```php
$isValidEmailCallback = Validation::createIsValidCallable(new Email(mode: Email::VALIDATION_MODE_HTML5));
$isValidEmailCallback('example@example.com');
```

### Additional Context

https://symfony.com/doc/current/validation.html#validation-callables

### Related issues
- https://github.com/symfony/symfony/issues/49355#issuecomment-1427990490

**Comment by MatTheCat** [comment]
As you can pass your own validator to `createIsValidCallable`, the workaround is essentially the same:

```php
$isValidEmailCallback = Validation::createIsValidCallable(
    Validation::createValidatorBuilder()
        ->setConstraintValidatorFactory(new ConstraintValidatorFactory([EmailValidator::class => new EmailValidator(Email::VALIDATION_MODE_HTML5)]))
        ->getValidator(),
    new Email(mode: Email::VALIDATION_MODE_HTML5)
);
```

**Comment by mhujer** [comment]
@MatTheCat thanks for the workaround!

**Comment by MatTheCat** [comment]
@mhujer you’re welcome! I guess this issue can be closed as #48644 says creating a custom validator is the way to go until v7.
